Build failure on platform `android` (within Termux)

When running `npm install` on the repo, building the native code fails with the following stacktrace:

```
/data/data/com.termux/files/home/apps/node-datachannel/src/peer-connection-wrapper.cpp:523:47: error: no matching conversion for functional-style cast from 'Napi::Number' to 'std::chrono::milliseconds' (aka 'duration<long long, ratio<1LL, 1000LL>>')
                    init.reliability.rexmit = std::chrono::milliseconds(reliability.Get("rexmit").As<Napi::Number>());
                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/data/com.termux/files/usr/include/c++/v1/chrono:1021:28: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'Napi::Number' to 'const std::chrono::duration<long long, std::ratio<1, 1000>>' for 1st argument
class _LIBCPP_TEMPLATE_VIS duration                                                                                 ^
/data/data/com.termux/files/usr/include/c++/v1/chrono:1021:28: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'Napi::Number' to 'std::chrono::duration<long long, std::ratio<1, 1000>>' for 1st argument
/data/data/com.termux/files/usr/include/c++/v1/chrono:1073:18: note: candidate template ignored: requirement 'is_convertible<Napi::Number, long long>::value' was not satisfied [with _Rep2 = Napi::Number]
        explicit duration(const _Rep2& __r,
                 ^
/data/data/com.termux/files/usr/include/c++/v1/chrono:1085:9: note: candidate template ignored: could not match 'duration<_Rep2, _Period2>' against 'Napi::Number'
        duration(const duration<_Rep2, _Period2>& __d,
        ^
/data/data/com.termux/files/usr/include/c++/v1/chrono:1066:9: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
        duration() = default;
        ^
1 error generated.
make[2]: *** [CMakeFiles/node_datachannel.dir/build.make:174: CMakeFiles/node_datachannel.dir/src/peer-connection-wrapper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:316: CMakeFiles/node_datachannel.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

Unfortunately my knowledge with C++ is very limited so I'm not sure what causes this.

Node version: v16.18.1
npm version: 6.14.17
libc++ version: 25b

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,10 @@ if(WIN32)
     set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 endif()
 
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Android")
+    set(CMAKE_C_FLAGS "-Wno-error=unused-but-set-variable -Wno-error=strict-prototypes")
+endif()
+
 # Got linker error for arm64
 # /usr/lib/gcc-cross/aarch64-linux-gnu/9/../../../../aarch64-linux-gnu/bin/ld: ../sysroot/usr/lib/aarch64-linux-gnu/libcrypto.a(sha1-armv8.o): relocation R_AARCH64_PREL64 against symbol `OPENSSL_armcap_P' which may bind externally can not be used when making a shared object; recompile with -fPIC
 # ../sysroot/usr/lib/aarch64-linux-gnu/libcrypto.a(sha1-armv8.o): in function `sha1_block_armv8':
@@ -85,6 +89,8 @@ set(LINK_LIBRARIES
 
 if(APPLE)
     #
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Android")
+    list(APPEND LINK_LIBRARIES -static-libgcc)
 elseif(UNIX)
     list(APPEND LINK_LIBRARIES -static-libgcc -static-libstdc++)
 endif()

--- a/src/peer-connection-wrapper.cpp
+++ b/src/peer-connection-wrapper.cpp
@@ -517,10 +517,10 @@ Napi::Value PeerConnectionWrapper::createDataChannel(const Napi::CallbackInfo &i
                 switch (reliability.Get("type").As<Napi::Number>().Uint32Value())
                 {
                 case 1:
-                    init.reliability.rexmit = reliability.Get("rexmit").As<Napi::Number>();
+                    init.reliability.rexmit = reliability.Get("rexmit").As<Napi::Number>().Uint32Value();
                     break;
                 case 2:
-                    init.reliability.rexmit = std::chrono::milliseconds((int)reliability.Get("rexmit").As<Napi::Number>().ToNumber());
+                    init.reliability.rexmit = std::chrono::milliseconds(reliability.Get("rexmit").As<Napi::Number>().Uint32Value());
                     break;
                 }
             }

--- a/src/peer-connection-wrapper.cpp
+++ b/src/peer-connection-wrapper.cpp
@@ -520,7 +520,7 @@ Napi::Value PeerConnectionWrapper::createDataChannel(const Napi::CallbackInfo &i
                     init.reliability.rexmit = reliability.Get("rexmit").As<Napi::Number>();
                     break;
                 case 2:
-                    init.reliability.rexmit = std::chrono::milliseconds(reliability.Get("rexmit").As<Napi::Number>());
+                    init.reliability.rexmit = std::chrono::milliseconds((int)reliability.Get("rexmit").As<Napi::Number>().ToNumber());
                     break;
                 }
             }

--- a/src/peer-connection-wrapper.cpp
+++ b/src/peer-connection-wrapper.cpp
@@ -517,7 +517,7 @@ Napi::Value PeerConnectionWrapper::createDataChannel(const Napi::CallbackInfo &i
                 switch (reliability.Get("type").As<Napi::Number>().Uint32Value())
                 {
                 case 1:
-                    init.reliability.rexmit = reliability.Get("rexmit").As<Napi::Number>().Uint32Value();
+                    init.reliability.rexmit = static_cast<int>(reliability.Get("rexmit").As<Napi::Number>().ToNumber());
                     break;
                 case 2:
                     init.reliability.rexmit = std::chrono::milliseconds(reliability.Get("rexmit").As<Napi::Number>().Uint32Value());

